### ISSUE TITLE
Error: unknown JSON.parse problem when firebase storage deploy

### ISSUE THREAD

**Issue by Nifu-Uchi** [firebase/firebase-tools#4117]
### [REQUIRED] Environment info

**firebase-tools:v10.1.3

**Platform:Github Actions, macOS local

### [REQUIRED] Test case

- do `firebase deploy --force --only storage --project ${{ env.PROJECT_ID }}`

### [REQUIRED] Steps to reproduce

- do `firebase deploy --force --only storage --project ${{ env.PROJECT_ID }}`
- its occurred in local command and Github workflow.
-  And it was not happend before the Latest `firebase-tools` update for v10.1.3
### [REQUIRED] Expected behavior

- firebase storage deploy success

### [REQUIRED] Actual behavior

- it shows JSON.parse ERROR.
```
i  deploying storage, firestore


There was an issue deploying your functions. Verify that your project has a Google App Engine instance setup at https://console.cloud.google.com/appengine and try again. If this issue persists, please contact support.

Error: Unable to parse JSON: SyntaxError: Unexpected token < in JSON at position 0
Error: Process completed with exit code 1.
```
- This error message seems to be derived from the code [here](https://github.com/firebase/firebase-tools/compare/v10.1.2...v10.1.3#diff-041ee7faf505a1f34aaf8d58f00ba9ccad426e4a2dcc9b84a0a106ff8e2874b7R411-R413)
- Maybe the content of `text` has become `<anonymous>` or `<undefined>`?.

**Comment by nvme0** [comment]
Can confirm I'm having this issue as well. Works fine on 10.1.2, but breaks on 10.1.3.

```
=== Deploying to '***'...

i  deploying storage


There was an issue deploying your functions. Verify that your project has a Google App Engine instance setup at https://console.cloud.google.com/appengine and try again. If this issue persists, please contact support.

Error: Unable to parse JSON: SyntaxError: Unexpected token < in JSON at position 0
```

**Comment by ricardodebeijer** [comment]
Can also confirm this, both in our CI and locally, both use 10.1.3. Attached is the `firebase-debug.log` with the  `--debug` flag
[firebase-debug.log](https://github.com/firebase/firebase-tools/files/7978003/firebase-debug.log)

**Comment by Nifu-Uchi** [comment]
@ricardodebeijer Thank You!
It looks like this API  returned `none` caused the error.
```
[debug] [2022-02-01T11:27:49.262Z] >>> [apiv2][query] GET https://storage.googleapis.com/v1/apps/mobilenext-qa [none]
```

**Comment by aezur** [comment]
Same thing but with functions.

**Comment by bkendall** [comment]
This should be fixed in `v10.1.4`. Thanks again for raising, and sorry for the trouble. Added a storage deploy test too so that at least _that_ path would catch issues in the future 🙂
